sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/ResourceExtensions.cs

```diff
@@ -9,90 +9,120 @@
 using Azure.Monitor.OpenTelemetry.Exporter.Models;
 using OpenTelemetry.Resources;
 
-namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals;
+
+internal static class ResourceExtensions
 {
-    internal static class ResourceExtensions
-    {
-        private const string s_AiSdkPrefixKey = "ai.sdk.prefix";
-        private const int Version = 2;
+    private const string AiSdkPrefixKey = "ai.sdk.prefix";
+    private const string DefaultServiceName = "unknown_service";
+    private const int Version = 2;
 
-        internal static AzureMonitorResource? CreateAzureMonitorResource(this Resource resource, string? instrumentationKey = null)
+    internal static AzureMonitorResource? CreateAzureMonitorResource(this Resource resource, string? instrumentationKey = null)
+    {
+        if (resource == null)
         {
-            if (resource == null)
-            {
-                return null;
-            }
+            return null;
+        }
 
-            AzureMonitorResource azureMonitorResource = new AzureMonitorResource();
-            MetricsData? metricsData = null;
-            string? serviceName = null;
-            string? serviceNamespace = null;
+        AzureMonitorResource azureMonitorResource = new AzureMonitorResource();
+        MetricsData? metricsData = null;
+        AksResourceProcessor? aksResourceProcessor = null;
+        string? serviceName = null;
+        string? serviceNamespace = null;
+        string? serviceInstance = null;
+        bool? hasDefaultServiceName = null;
```

Comment:
this is nullable, but i don't see the null value used. I think it would be safe to change this to `bool hasDefaultServiceName = false`

Reply:
used indirectly here - 
```
            if (hasDefaultServiceName != false && aksRoleName != null)
            {
                azureMonitorResource.RoleName = aksRoleName;
            }
```

Reply:
Null satisfies "If service.name does not exist" condition.

Reply:
Not a strong preference but we could do `if ((hasDefaultServiceName || serviceName == null) && aksRoleName != null)` to make it more readable.

Reply:
Not sure if I followed this^. With the condition I was suggesting the boolean will not be nullable.

Reply:
In a normal scenario, people may not use '==' or '!=' with Booleans. This syntax is typically used only with nullable Booleans. As there is no strong preference, I would like to keep it this way.